### ISSUE TITLE
Cannot find module kind-of

### ISSUE THREAD

**Issue by kumavis** [jonschlinkert/merge-deep#18]
encountered this problem with fresh node_modules

bundling with webpack against `merge-deep@3.0.3`

relevant issue with recent reports https://github.com/jonschlinkert/merge-deep/issues/4

**Comment by kumavis** [comment]
looks like `webpack@4.37.0` doesn't know how to parse `clone-deep@0.2.4`
```
WARNING in /home/xyz/Development/lavamoat/node_modules/clone-deep/utils.js 7:34-41
Critical dependency: require function is used in a way in which dependencies cannot be statically extracted
 @ /home/xyz/Development/lavamoat/node_modules/clone-deep/index.js
 @ ../core/node_modules/merge-deep/index.js
 @ ../core/src/mergeConfig.js
 @ ./src/App.js
 @ ./src/index.js
```

**Comment by kumavis** [comment]
due to this lazy cache mechanism https://npmfs.com/package/clone-deep/0.2.4/utils.js#L10

**Comment by kumavis** [comment]
relevant PRs https://github.com/jonschlinkert/merge-deep/pull/12 https://github.com/jonschlinkert/merge-deep/pull/14 https://github.com/jonschlinkert/merge-deep/pull/9

**Comment by kumavis** [comment]
@doowb more info over here :+1:

**Comment by doowb** [comment]
@kumavis :) I just saw these... thank you for more information.

Have you tried [unlazy-loader](https://github.com/doowb/unlazy-loader)? It might be useful until we update the dependencies using `lazy-cache`.

**Comment by kumavis** [comment]
fwiw just tested the latest versions and they worked with the current `merge-deep` source for my basic json merging needs
- `arr-union@3.1.0`
- `clone-deep@4.0.1`
- `kind-of@6.0.3`

not sure what breaking changes bubble up but could always just udpate them all and cut a major :shrug:

**Comment by doowb** [comment]
Yes... I believe the path will be to update the dependencies and publish a new major version.

Most of the breaking changes have to do with Node.js compatibility... this is currently set at `>=0.10.0`, and will need to be bumped to at least what `clone-deep` is using (`>=6`) because of the use of `const`.

**Comment by gtsigner** [comment]
also same error invoke when using puppeteer-extra-plugin
![image](https://user-images.githubusercontent.com/13638996/114993648-8ab9b080-9e8b-11eb-824a-b2e09611e51a.png)

it still dosen't work after using unlazy-loader

**Comment by quroro** [comment]
Can anyone help me with this problem?
I'm stuck.

**Comment by mycoin** [comment]
USE assign-deep (https://www.npmjs.com/package/assign-deep) instand !

**Comment by JasonHoku** [comment]
Hello!

If you're using webpack and electron I found a fix for this by making sure to include any merge-deep dependencies as an external and to declare the node_modules path relative to your package.json in your package.json as an 'extraResource'.
